src/utils.ts

```diff
@@ -98,3 +99,35 @@ export function sortTheThings(
   }
   return -1;
 }
+export async function execWithOutput(
+  cmd: Parameters<typeof exec>[0],
+  args: Parameters<typeof exec>[1],
+  opts?: { notrim?: boolean; cwd?: string }
+) {
+  let stdout = "";
+  let stderr = "";
+
+  const options: ExecOptions = {
+    listeners: {
+      stdout: (data: Buffer) => {
+        stdout += data.toString();
+      },
+      stderr: (data: Buffer) => {
+        stderr += data.toString();
+      },
+    },
+    cwd: opts?.cwd,
+  };
+
+  const exitCode = await exec(cmd, args, options);
+  if (exitCode !== 0) {
+    throw Error(stderr);
+  }
+
+  if (!opts?.notrim) {
+    stdout = stdout.trim();
+    stderr = stderr.trim();
```

Comment:
this is redundant as it never gets called after the Error above?

Reply:
It could exit with status code 0 but have used `stderr`.